mirrord/agent/src/cgroup.rs

```diff
@@ -0,0 +1,262 @@
+//! Logic for pausing using cgroup directly
+//! Pause requires privileged - assumes ephemeral (hardcoded pid 1)
+use std::path::{Path, PathBuf};
+
+use nix::mount::{mount, MsFlags};
+use tokio::{
+    fs::{create_dir, try_exists, File, OpenOptions},
+    io::{AsyncBufReadExt, AsyncWriteExt, BufReader},
+};
+use tracing::{trace, warn};
+
+use crate::{
+    error::{AgentError, Result},
+    namespace::{set_namespace, NamespaceType},
+};
+
+const CGROUP_MOUNT_PATH: &str = "/mirrord_cgroup";
+const CGROUP_SUBGROUP_MOUNT_PATH: &str = "/mirrord_cgroup/subgroup";
+const CGROUPV2_PROCS_FILE: &str = "cgroup.procs";
```

Comment:
Are associated constants still unstable? If so can leave it like this, otherwise I would suggest putting the v1/v2 specific consts in the impl blocks:

```rust
impl CgroupV1 {
    const CGROUP_V1_FREEZE_PATH: ....;
}
```